lib/logstash/pluginmanager/util.rb

```diff
@@ -1,64 +1,101 @@
+module LogStash::PluginManager
 
-class LogStash::PluginManager::Util
-
-  def self.logstash_plugin?(gem)
+  # check for valid logstash plugin gem name & version or .gem file, logs errors to $stdout
+  # uses Rubygems API and will remotely validated agains the current Gem.sources
+  # @param plugin [String] plugin name or .gem file path
+  # @param version [String] gem version requirement string
+  # @return [Boolean] true if valid logstash plugin gem name & version or a .gem file
+  def self.is_logstash_plugin?(plugin, version = nil)
+    if is_plugin_file?(plugin)
+      begin
+        return is_logstash_plugin_gem_spec?(plugin_file_spec(plugin))
+      rescue => e
+        $stderr.puts("Error reading plugin file #{plugin}, caused by #{e.class}")
```

Comment:
We have lots of $stderr, though I wish for using Cabin (or a logger, in general) instead.